### PR TITLE
feat: watering interaction + i18n

### PR DESCRIPTION
**Pull request by Jaszkowic**

### CODE REVIEW

File: src/components/tree-detail/watering-dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        <div className="flex flex-col gap-2">
+          <div className="text-lg font-semibold">
+            {i18n.treeDetail.waterNeed.wateredWhen}
+          </div>
+          <input
+            className="rounded-lg border-2 p-4"
+            type="datetime-local"
+            value={format(parseISO(waterDate.toISOString()), dateFormat)}
+            onChange={(e) => {
+              const parsedDate = parse(e.target.value, dateFormat, new Date());
```
Comment (ff6347) [inline src/components/tree-detail/watering-dialog.tsx:54]: What does parse do if the parse fails?

**Review by ff6347 — CHANGES_REQUESTED** [review]

**Comment by Jaszkowic** [comment]
@ff6347 I will handle the errors with the global error handler introduced in commit https://github.com/technologiestiftung/giessdenkiez-de/pull/734/commits/8c4985b101f2fcd53cb330f8a837a8e3a1acd26e once the PR https://github.com/technologiestiftung/giessdenkiez-de/pull/734 is approved and merged

**Comment by ff6347** [comment]
> src/components/tree-detail/hooks/use-fetch-tree-watering-data.tsx(47,1): error TS1005: ')' expected.
> Error: Command "npm run build" exited with 2

File: src/components/tree-detail/hooks/use-fetch-tree-watering-data.tsx [hunk 2]
```diff
@@ -10,36 +15,32 @@ export function useFetchTreeWateringData(
...
+        },
+      });
+      if (!res.ok) {
+        return;
       }
-    };
+      const json = await res.json();
+      setTreeWateringData(json.data);
+    } catch (_) {
+      setTreeWateringData([]);
```
[the remark this answers is not in the document]
Comment (Jaszkowic) [reply src/components/tree-detail/hooks/use-fetch-tree-watering-data.tsx]: fixed in https://github.com/technologiestiftung/giessdenkiez-de/pull/735/commits/1e1fdd0046d5f9d063b7047d08bd487ef9b55e8e

File: src/components/tree-detail/hooks/use-water-tree.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+          timestamp: date.toISOString(),
+        }),
+        headers: {
+          Authorization: `Bearer ${access_token}`,
+          "Content-Type": "application/json",
+        },
+        signal: abortController.signal,
+      });
+      if (!res.ok) {
+        setWateringLoading(false);
```
[the remark this answers is not in the document]
Comment (Jaszkowic) [reply src/components/tree-detail/hooks/use-water-tree.tsx:42]: fixed in https://github.com/technologiestiftung/giessdenkiez-de/pull/735/commits/1e1fdd0046d5f9d063b7047d08bd487ef9b55e8e

File: src/components/tree-detail/watering-dialog.tsx (hunk 1 above)
Comment (Jaszkowic) [reply src/components/tree-detail/watering-dialog.tsx:54]: Not sure if it is realistic that it fails, it does not parse a user input, but parses the value that is returned by the "datetime-local" managed input. Can you find a scenario where it fails?

**Comment by Jaszkowic** [comment]
> > src/components/tree-detail/hooks/use-fetch-tree-watering-data.tsx(47,1): error TS1005: ')' expected.
> > Error: Command "npm run build" exited with 2

@ff6347 merge conflicts, fixed in last commit

File: src/components/tree-detail/hooks/use-fetch-tree-watering-data.tsx [hunk 4]
```diff
@@ -10,36 +15,32 @@ export function useFetchTreeWateringData(
...
+      const geocodingUrl = `${import.meta.env.VITE_API_ENDPOINT}/get/lastwatered?id=${treeData.id}`;
+      const res = await fetch(geocodingUrl, {
+        method: "GET",
+        headers: {
+          Authorization: `Bearer ${import.meta.env.VITE_ANON_KEY}`,
+          "Content-Type": "application/json",
+        },
+      });
+      if (!res.ok) {
+        return;
```
[the remark this answers is not in the document]
Comment (Jaszkowic) [reply src/components/tree-detail/hooks/use-fetch-tree-watering-data.tsx:38]: unified pattern introduced in last commit

File: src/components/tree-detail/watering-dialog.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+          <SecondaryButton
+            label={i18n.treeDetail.waterNeed.waterCancel}
+            disabled={false}
+            onClick={close}
+          />
+          <PrimaryButton
+            label={i18n.treeDetail.waterNeed.waterSave}
+            disabled={false}
+            onClick={async () => {
+              await waterTree(amount, waterDate);
```
[the remark this answers is not in the document]
Comment (Jaszkowic) [reply src/components/tree-detail/watering-dialog.tsx:69]: fixed in last commit

**Review by ff6347 — APPROVED** [review]

**Pull request merged**